### PR TITLE
[ci] Add coq-lsp to CI

### PR DESCRIPTION
**Pull request by ejgallego**

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -870,6 +871,14 @@ plugin:ci-serapi:
   needs:
   - build:edge+flambda
   - library:ci-mathcomp
+  only: *full-ci
+
+plugin:ci-coq_lsp:
+  extends: .ci-template-flambda
+  needs:
+  - build:edge+flambda
+  - plugin:ci-serapi
```
Comment (SkySkimmer) [inline .gitlab-ci.yml:880]:
```suggestion
  - library:ci-mathcomp
  - plugin:ci-serapi
```

Comment (ejgallego) [reply .gitlab-ci.yml:880]: coq-lsp test suite doesn't need math-comp, so I tried to be more economical.

**Comment by ejgallego** [comment]
My hack to avoid full_ci didn't work

File: .gitlab-ci.yml (hunk 1 above)
Comment (SkySkimmer) [reply .gitlab-ci.yml:880]: The CI is going to do `make ci-coq_lsp`, which transitively depends on mathcomp in Makefile.ci, so it's going to build mathcomp.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -715,6 +715,7 @@ library:ci-math_classes:
 
 library:ci-mathcomp:
   extends: .ci-template-flambda
+  only: *full-ci
```
Comment (SkySkimmer) [inline .gitlab-ci.yml:718]: if you want to enable the job on light ci I think this works
```suggestion
  variables:
    FULL_CI: "true"
```

**Comment by ejgallego** [comment]
@coqbot run full CI

**Comment by ejgallego** [comment]
Seems to work!

**Review by silene — CHANGES_REQUESTED** [review]
I am really confused about the dependency of `plugin:ci-coq_lsp` on `library:ci-mathcomp`. Given the discussion, it seems to be on purpose, but I have no idea what it achieves. At the very least, there should be a comment explaining why this dependency exists.

**Comment by ejgallego** [comment]
Good catch @silene , that dependency was for the test suite, it is not needed now; let me remove it.

**Comment by silene** [comment]
There is nothing in the log that shows anything being built for `coq-lsp` (contrarily to other plugins, e.g., `coq-serapi`). Could you increase the verbosity of the build or something?

**Comment by ejgallego** [comment]
> There is nothing in the log that shows anything being built for `coq-lsp` (contrarily to other plugins, e.g., `coq-serapi`). Could you increase the verbosity of the build or something?

Good idea, I've added a call to `coq-lsp --version` which should show the binary is there.

**Comment by ejgallego** [comment]
> Good idea, I've added a call to `coq-lsp --version` which should show the binary is there.

Seems to work!

**Review by silene — APPROVED** [review]

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Pull request merged**